R/patient_profile_charts.R

```diff
@@ -358,3 +358,52 @@ drawDemoTable <- function(adlb, settings, demo_vars = c("USUBJID", "AGE", "SEX",
     gt()
 
 }
+
+#' Draw BUN/serum creatinine over time,
+#'
+#' @param adlb lab data in tall format that must contain DY for study day,
+#'   VISITN for visit number, TEST for lab test, and STRESN for lab value
+#' @param settings settings object with column mappings
+#'
+#' @import ggplot2
+#' @import dplyr
+#' @import RColorBrewer
+#' @importFrom plotly ggplotly
+#' @importFrom plotly config
+#' @return ggplot object
+drawBunCreat <- function(adlb, settings) {
+  adlb_norm <- adlb %>%
+    filter(.data[[settings$measure_col]] == settings$measure_values[["BUN/CREAT"]])
+  
+  uacr_unit <- unique(adlb_norm[[settings$unit_col]])
```

Comment:
I'm thinking this unit variabel name should be different since no uacr involved, also I think on safetygraphics app example it gives an error for this chart because there's no unit, we may want to cahnge this to check for units because the if(length) statement below crashes it

Reply:
Thanks Preston - uacr_unit references removed.